Source/WebCore/html/CanvasBase.cpp

```diff
@@ -48,16 +49,24 @@
 #include <JavaScriptCore/JSLock.h>
 #include <atomic>
 #include <wtf/Vector.h>
+#include <wtf/WeakRandom.h>
 #include <wtf/text/MakeString.h>
 
 namespace WebCore {
 
 constexpr InterpolationQuality defaultInterpolationQuality = InterpolationQuality::Low;
 static std::optional<size_t> maxCanvasAreaForTesting;
 
-CanvasBase::CanvasBase(IntSize size, const ScriptExecutionContext& context)
+static std::optional<uint64_t> needsCanvasNoiseHashSalt(ScriptExecutionContext& context)
```

Comment:
Why not `static bool needsCanvasNoiseHashSalt(ScriptExecutionContext&);`?

Reply:
Or I guess you could do something like:


```cpp
static std::optional<uint64_t> canvasNoiseHashSaltIfNeeded(ScriptExecutionContext& context)
{
    auto policies = context.noiseInjectionPolicies();
    if (policies.contains(NoiseInjectionPolicy::Minimal)
        || (policies.contains(NoiseInjectionPolicy::Enhanced) && context.requiresScriptExecutionTelemetry(ScriptTelemetryCategory::Canvas)))
        return context.noiseInjectionHashSalt();
    return { };
}

...

, m_canvasNoiseHashSalt { canvasNoiseHashSaltIfNeeded(context) }
```

Reply:
🤦 Oops, yes — I meant for this to return a `bool` in this patch (I was torn between returning a `bool` or `std::optional<uint64_t>` and somehow ended up with both).